test/integration/modules/edgecore/eventbus/eventbus_test.go

```diff
@@ -0,0 +1,204 @@
+		})
+
+		It("TC_TEST_EBUS_3: Sending data to device twin module", func() {
+			if Token_client = Client.Publish(DeviceTwinUpdate, 0, false, "messagetoDevice_Twin_update"); Token_client.Wait() && Token_client.Error() != nil {
+				common.Failf("client.Publish() Error is %s", Token_client.Error())
+			} else {
+				common.InfoV6("client.Publish Success !!")
+			}
+			Expect(Token_client.Error()).NotTo(HaveOccurred())
+			Client.Disconnect(1)
+		})
+
+		It("TC_TEST_EBUS_4: Sending data to membership module", func() {
+			if Token_client = Client.Publish(DeviceMembershipUpdate, 0, false, "messagetoDevice_Membership_update"); Token_client.Wait() && Token_client.Error() != nil {
+				common.Failf("client.Publish() Error is %s", Token_client.Error())
+			} else {
+				common.InfoV6("client.Publish Success !!")
+			}
+			Expect(Token_client.Error()).NotTo(HaveOccurred())
+			Client.Disconnect(1)
+		})
+
+		It("TC_TEST_EBUS_5: Sending data to device module", func() {
+			if Token_client = Client.Publish(DeviceUpload, 0, false, "messagetoDevice_upload"); Token_client.Wait() && Token_client.Error() != nil {
+				common.Failf("client.Publish() Error is %s", Token_client.Error())
+			} else {
+				common.InfoV6("client.Publish Success !!")
+			}
+			Expect(Token_client.Error()).NotTo(HaveOccurred())
+			Client.Disconnect(1)
+		})
+
+		It("TC_TEST_EBUS_6: change the device status to online from eventbus", func() {
+			var message DeviceUpdate
+			message.State = "online"
+			body, _ := json.Marshal(message)
+			deviceID := ctx.Cfg.DeviceID
+
+			if Token_client = Client.Publish(dtcommon.DeviceETPrefix+deviceID+dtcommon.DeviceETStateUpdateSuffix, 0, false, body); Token_client.Wait() && Token_client.Error() != nil {
+				common.Failf("client.Publish() Error is %s", Token_client.Error())
+			} else {
+				common.InfoV6("client.Publish Success !!")
+			}
+			Expect(Token_client.Error()).NotTo(HaveOccurred())
+			time.Sleep(10 * time.Second)
+			DeviceState := getDeviceStateFromDB(deviceID)
+			Expect(DeviceState).Should(Equal("online"))
+			Client.Disconnect(1)
+		})
+
+		It("TC_TEST_EBUS_7: change the device status to unknown from eventbus", func() {
+			var message DeviceUpdate
+			message.State = "unknown"
+			body, _ := json.Marshal(message)
+			deviceID := ctx.Cfg.DeviceID
+
+			if Token_client = Client.Publish(dtcommon.DeviceETPrefix+deviceID+dtcommon.DeviceETStateUpdateSuffix, 0, false, body); Token_client.Wait() && Token_client.Error() != nil {
+				common.Failf("client.Publish() Error is %s", Token_client.Error())
+			} else {
+				common.InfoV6("client.Publish Success !!")
+			}
+			Expect(Token_client.Error()).NotTo(HaveOccurred())
+			time.Sleep(10 * time.Second)
+			DeviceState := getDeviceStateFromDB(deviceID)
+			Expect(DeviceState).Should(Equal("unknown"))
+			Client.Disconnect(1)
+		})
+
+		It("TC_TEST_EBUS_8: change the device status to offline from eventbus", func() {
+			var message DeviceUpdate
+			var deviceID string
```

Comment:
can't we directly do deviceID := ctx.Cfg.DeviceID instead of this additional declaration ?

Reply:
@rohitsardesai83 , deviceID variable is removed from all the tests, replaced with global varibale to carry the DeviceID maintained accross the modules (testManager/eventbus/devicetwin). now DeviceID is generated via random string.